File: scala/akka/hello_caseclass/src/main/scala/fits.sample/SampleActor.scala
package fits.sample

import akka.actor._

case class Hello(message: String)
case class Hi(message: String)

class SampleActor extends Actor {
	def receive = {
		case he: Hello => log.info("*** received Hello : " + he.message)
		case hi: Hi => log.info("*** received Hi : " + hi.message)
		case _ => {
			log.info("*** received others")
			//reply
			self.reply("return")
		}
	}
}

File: java/akka/actor_osgi_sample/src/main/java/sample/SampleActor.java
package sample;

import akka.actor.*;

public class SampleActor extends UntypedActor {
	@Override
	public void preStart() {
		System.out.println("preStart");
	}

	@Override
	public void onReceive(Object msg) {
		if (msg instanceof String) {
			System.out.println("*** " + msg);
		}
		else {
			unhandled(msg);
		}
	}
}
